Explorer/Assets/DCL/Infrastructure/ECS/SceneLifeCycle/IncreasingRadius/ResolveSceneStateByIncreasingRadiusSystem.cs

```diff
@@ -96,117 +161,268 @@ private void ProcessScenesUnloadingInRealm()
-        private void ProcessesFixedRealm([Data] float maxLoadingSqrDistance, ref RealmComponent realmComponent, ref FixedScenePointers fixedScenePointers)
+        private void ProcessesFixedRealm(in RealmComponent realmComponent, ref FixedScenePointers fixedScenePointers)
         {
             if (fixedScenePointers.AllPromisesResolved)
-                StartScenesLoading(ref realmComponent, maxLoadingSqrDistance);
+                StartScenesLoading(realmComponent);
         }
 
-        private void StartScenesLoading(ref RealmComponent realmComponent, float maxLoadingSqrDistance)
+        private int lengthWhenSortingStarted;
+
+        private void StartScenesLoading(in RealmComponent realmComponent)
         {
             if (sortingJobHandle is { IsCompleted: true })
             {
                 sortingJobHandle.Value.Complete();
-                CreatePromisesFromOrderedData(realmComponent.Ipfs);
+
+                // Since adding new values is throttled, arrays may be out of sync. They need to be synced to work
+                if (arraysInSync)
+                    CreatePromisesFromOrderedData(realmComponent.Ipfs);
             }
 
             if (sortingJobHandle is { IsCompleted: false }) return;
 
-            // Start new sorting
-            // Order the scenes definitions by the CURRENT partition and serve first N of them
-
-            orderedData.Clear();
+            Vector2Int currentParcel = playerTransform.position.ToParcel();
+            int xCoordinate = currentParcel.x;
+            int yCoordinate = currentParcel.y;
 
-            foreach (ref Chunk chunk in World.Query(in START_SCENES_LOADING))
+            unsafe
             {
-                ref Entity entityFirstElement = ref chunk.Entity(0);
-                ref PartitionComponent partitionComponentFirst = ref chunk.GetFirst<PartitionComponent>();
+                OrderedDataNative* dataPtr = orderedDataNative.GetUnsafePtr();
 
-                foreach (int entityIndex in chunk)
+                for (var i = 0; i < orderedDataManaged.Count; i++)
                 {
-                    ref readonly Entity entity = ref Unsafe.Add(ref entityFirstElement, entityIndex);
-                    ref PartitionComponent partitionComponent = ref Unsafe.Add(ref partitionComponentFirst, entityIndex);
-
-                    if (partitionComponent.RawSqrDistance >= maxLoadingSqrDistance) continue;
+                    OrderedDataManaged currentOrderedData = orderedDataManaged[i];
+                    currentOrderedData.UpdatePlayerInParcel(xCoordinate, yCoordinate);
 
-                    orderedData.Add(new OrderedData
+                    dataPtr[i] = new OrderedDataNative
                     {
-                        Entity = entity,
-                        Data = new DistanceBasedComparer.DataSurrogate(partitionComponent.RawSqrDistance, partitionComponent.IsBehind),
-                    });
+                        ReferenceListIndex = i,
```

Comment:
Is not it the same index as the index in the managed array so we don't need to store it? (As we always re-sort the whole array, right?)

Reply:
Whats being sorted is the native array, the managed array stays the same. I did it like this because sorting the native array with a job was more performant than sorting the managed array. 

So, they are not the same index. The managed one stays unsorted, while the native one is sorted

Reply:
Yes, makes sense!